lightningd/onion_message.c

```diff
@@ -432,7 +432,12 @@ static struct command_result *json_send_onion_message(struct command *cmd,
 				hops[i].rawtlv, tal_bytelen(hops[i].rawtlv));
 		sphinx_add_hop(sphinx_path, &hops[i].id, take(tlv_with_len));
 	}
+	/* BOLT-offers #4:
+	 * - SHOULD set `len` to 1366 or 32834.
+	 */
 	op = create_onionpacket(tmpctx, sphinx_path, ROUTING_INFO_SIZE, &path_secrets);
+	if (!op)
+		op = create_onionpacket(tmpctx, sphinx_path, 32768, &path_secrets);
```

Comment:
Technically we could compute the payload size using `sphinx_path_payloads_size` to determine which format will be required:

https://github.com/ElementsProject/lightning/blob/25463bd5a755d9617c262dbbe1827d1097b0d969/common/sphinx.c#L491-L492

This would save us two allocations and de-allocations:

https://github.com/ElementsProject/lightning/blob/25463bd5a755d9617c262dbbe1827d1097b0d969/common/sphinx.c#L502-L506